### PR TITLE
Allow internal shares to have a default expiration date

### PR DESCRIPTION
**Pull request by rullzer**
TODO:
* [x] Needs admin UI
  * [x] basics
  * [x] Fix the js so it opens etc.
* [x] Needs more checks
* [X] Expose variables
* [x] Need integration of enforcement and max date into the sidebar
* [x] Actually expire the share
* [x] Expose in capabilities

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: apps/settings/templates/settings/admin/sharing.php [hunk 1]
```diff
@@ -37,6 +37,22 @@
 			   value="1" <?php if ($_['shareAPIEnabled'] === 'yes') print_unescaped('checked="checked"'); ?> />
 		<label for="shareAPIEnabled"><?php p($l->t('Allow apps to use the Share API'));?></label><br/>
 	</p>
+
+	<p id="internalShareSettings" class="indent" <?php if ($_['shareAPIEnabled'] === 'no') p('hidden'); ?>">
```
Comment (skjnldsv) [inline apps/settings/templates/settings/admin/sharing.php]:
```suggestion
	<p id="internalShareSettings" class="indent" <?php if ($_['shareAPIEnabled'] === 'no') p('hidden'); ?>>
```

**Review by skjnldsv — APPROVED** [review]

**Comment by rullzer** [comment]
Time for the reviews and to get in!

File: apps/settings/templates/settings/admin/sharing.php [hunk 2]
```diff
@@ -37,6 +37,22 @@
...
 	</p>
+
+	<p id="internalShareSettings" class="indent <?php if ($_['shareAPIEnabled'] === 'no') p('hidden'); ?>">
+		<input type="checkbox" name="shareapi_default_internal_expire_date" id="shareapiDefaultInternalExpireDate" class="checkbox"
+			   value="1" <?php if ($_['shareDefaultInternalExpireDateSet'] === 'yes') print_unescaped('checked="checked"'); ?> />
+		<label for="shareapiDefaultInternalExpireDate"><?php p($l->t('Set default expiration date for non-link shares'));?></label><br/>
+	</p>
+	<p id="setDefaultInternalExpireDate" class="double-indent <?php if ($_['shareDefaultInternalExpireDateSet'] === 'no' || $_['shareAPIEnabled'] === 'no') p('hidden');?>">
+		<?php p($l->t( 'Expire after ' )); ?>
+		<input type="text" name='shareapi_internal_expire_after_n_days' id="shareapiInternalExpireAfterNDays" placeholder="<?php p('7')?>"
```
Comment (ChristophWurst) [inline apps/settings/templates/settings/admin/sharing.php]: could be `type=number`

**Review by ChristophWurst — COMMENTED** [review]
Tested enforced expiration.

Works nicely for new new shares. For existing ones the expiration checkbox is now checked, but now date is set. Bug?

**Comment by rullzer** [comment]
> Tested enforced expiration.
> 
> Works nicely for new new shares. For existing ones the expiration checkbox is now checked, but now date is set. Bug?

I think that is the same on link shares.
So lets keep it for now :wink:

File: apps/settings/templates/settings/admin/sharing.php (hunk 2 above)
Comment (rullzer) [reply apps/settings/templates/settings/admin/sharing.php]: I would way we tackle all that (also for the other) if we finally move this to settings and non jquery magic ;)

**Comment by rullzer** [comment]
@ChristophWurst so... approve?

**Review by ChristophWurst — APPROVED** [review]

**Comment by rullzer** [comment]
Ah shit failing CI.
I'll fix that..

**Pull request merged**